File: ClimbEdge.Application/DTOs/AppUserDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClimbEdge.Application.DTOs
{
    public record class RegisterRequestDTO
    {
        public required string Email { get; set; }
        public required string Password { get; set; }
        public required string ConfirmPassword { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public RegisterRequestDTO() { }
    }

    public record class LoginRequestDTO
    {
        public required string Email { get; set; }
        public required string Password { get; set; }
    }

    public record class ChangePasswordRequestDTO
    {
        public required string CurrentPassword { get; set; }
        public required string NewPassword { get; set; }
        public required string ConfirmPassword { get; set; }
    }

    public record class ForgotPasswordRequestDTO
    {
        public required string Email { get; set; }
    }

    public record class ResetPasswordRequestDTO
    {
        public required string Email { get; set; }
        public required string Token { get; set; }
        public required string NewPassword { get; set; }
        public required string ConfirmPassword { get; set; }
    }

    public record class RefreshTokenRequestDTO
    {
        public required string Token { get; set; }
        public required string RefreshToken { get; set; }
    }

    public record class AuthResponseDTO
    {
        public required string Token { get; set; }
        public required string RefreshToken { get; set; }
        public required UserInfoDTO User { get; set; }
    }
    public record class UserInfoDTO : BaseDTO
    {
        public string? Email { get; set; }
        public string? UserName { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public bool EmailConfirmed { get; set; }
        public List<string>? Roles { get; set; }
    }
}

File: ClimbEdge.App/shared/src/types/AppUserDTO.ts
import { BaseDTO } from "./BaseDTO";

export interface RegisterRequestDTO {
  email: string;
  password: string;
  confirmPassword: string;
  firstName?: string;
  lastName?: string;
}

export interface LoginRequestDTO {
  email: string;
  password: string;
}

export interface ChangePasswordRequestDTO {
  currentPassword: string;
  newPassword: string;
  confirmPassword: string;
}

export interface ForgotPasswordRequestDTO {
  email: string;
}

export interface ResetPasswordRequestDTO {
  email: string;
  token: string;
  newPassword: string;
  confirmPassword: string;
}

export interface RefreshTokenRequestDTO {
  token: string;
  refreshToken: string;
}

export class UserInfoDTO extends BaseDTO {
  email?: string;
  userName?: string;
  firstName?: string;
  lastName?: string;
  emailConfirmed!: boolean;
  roles?: string[];
}

export class AuthResponseDTO {
  token!: string;
  refreshToken!: string;
  user!: UserInfoDTO;
}